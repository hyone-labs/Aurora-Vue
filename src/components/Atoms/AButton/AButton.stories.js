import { AButton } from '.';

export default {
  title: 'Design System/Atoms/AButton',
}

export const withText = () => ({
  components: { AButton },
  template: '<a-button icon="fas fa-user">Hello Button</a-button>',
})

export const variant = () => ({
  render() {
    return (
      <div>
        <AButton variant="primary">With JSX</AButton>
        <p />
        <AButton variant="secondary">With JSX</AButton>
        <p />
        <AButton variant="tertiary">With JSX</AButton>
        <p />
        <AButton variant="quaternary">With JSX</AButton>
      </div>
    );
  }
})

export const withIcon = () => ({
  render() {
    return (
      <div>
        <AButton icon="fas fa-user" icon-color="success">With JSX</AButton>
        <p />
        <AButton icon="fas fa-user" icon-color="warn">With JSX</AButton>
        <p />
        <AButton icon="fas fa-user" icon-color="danger" behavior="block">With JSX</AButton>
      </div>
    );
  }
})

export const isLoading = () => ({
  render() {
    return <AButton icon="fas fa-user" isLoading={true}>With JSX</AButton>;
  }
})

export const withBehavior = () => ({
  render() {
    return (
      <div>
        <AButton>💯 😀 😎 👍 💯</AButton>
        <p />
        <AButton behavior="block">💯 😀 😎 👍 💯</AButton>
      </div>
    );
  }
})
