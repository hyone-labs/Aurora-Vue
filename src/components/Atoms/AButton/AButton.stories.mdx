import { Meta, ArgsTable, Story, Canvas } from '@storybook/addon-docs/blocks';
import { action } from '@storybook/addon-actions'
import { linkTo } from '@storybook/addon-links'
import AButton from './AButton.vue';

<Meta title="MDX|Button" component={AButton} />

# Button

<ArgsTable of={AButton} />

This is a simple button with some text in it.

<Canvas>
  <Story name="With Text">
    {{
        components: { AButton },
        template: '<a-button @click="action">Hello Button</a-button>',
        methods: { action: action("clicked") }
    }}
  </Story>
</Canvas>

You can perform some action when the button is clicked.

<Canvas>
  <RAButton onClick={linkTo('Button', 'With Some Emoji')}>With JSX</RAButton>
</Canvas>

You can even have Emoji in the button.

<Canvas>
  <Story name="With Some Emoji">
    {{
        components: { AButton },
        template: '<a-button>😀 😎 👍 💯</a-button>'
    }}
  </Story>
</Canvas>
