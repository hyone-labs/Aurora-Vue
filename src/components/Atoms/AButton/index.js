import AButton from './AButton.vue';

export { AButton };