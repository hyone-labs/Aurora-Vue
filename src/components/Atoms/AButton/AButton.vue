<template>
  <button
    :class="{
      [`a-button`]: true,
      [`a-button--behavior-${behavior}`]: behavior,
      [`a-button--behavior-is-loading`]: isLoading,
      [`a-button--variant-${variant}`]: variant,
    }"
    @click="$emit('click')"
  >
    <AIcon :icon="icon" v-if="icon" :color="iconColor" class="a-button__icon"/>
    <AIcon icon="fas fa-circle-notch fa-spin" v-if="isLoading" class="a-button__icon a-button__icon--is-loading"/>
    <span
      :class="{
        'a-button__text': true,
        'a-button__text--icon': icon
      }"
    >
      <slot />
    </span>
  </button>
</template>

<script>
import { AIcon } from '@/components/Atoms/AIcon';

export default {
  name: 'a-button',
  components: { AIcon },
  props: {
    behavior: {
      type: String,
      default: '',
    },
    variant: {
      type: String,
      default: 'quaternary',
    },
    icon: {
      type: String,
      default: '',
    },
    iconColor: {
      type: String,
      default: '',
    },
    isLoading: {
      type: Boolean,
      default: false,
    },
  },
};
</script>

<style lang="scss" scoped>
.a-button {
  align-items: center;
  border-radius: var(--border-radius-normal);
  display: inline-flex;
  justify-content: center;
  padding: var(--size-small) var(--size-large);
  transition: background-color 250ms, opacity 250ms;
  font-weight: 500;
  min-height: 40px;
  position: relative;

  &--variant {
    &-primary {
      background-color: var(--colors-major-black);
      box-shadow: var(--shadow-elevation-02);
      color: var(--colors-original-white);
      &:hover {
        background-color: #CFDAE3;
      }
      &:active {
        background-color: #839fb6;
      }
    }

    &-secondary {
      background-color: var(--colors-original-white);
      box-shadow: var(--shadow-elevation-02);
      border: var(--size-micro) solid var(--colors-major-black);
      color: var(--colors-major-black);
      &:hover {
        background-color: #CFDAE3;
      }
      &:active {
        background-color: #839fb6;
      }
    }

    &-tertiary {
      background-color: transparent;
      color: var(--colors-major-black);
      &:hover {
        text-decoration: underline;
      }
      &:active {
        color: #839fb6;
        text-decoration: underline;
      }
    }

    &-quaternary {
      background-color: var(--colors-original-white);
      box-shadow: var(--shadow-elevation-02);
      color: var(--colors-normal-dark-blue);
      &:hover {
        background-color: #CFDAE3;
      }
      &:active {
        background-color: #839fb6;
      }
    }
  }

  &--behavior {
    &-block {
      width: 100%;
    }

    &-is-loading {
      & > .a-button__text {
        opacity: 0;
      }

      & > .a-button__icon {
        opacity: 0;
      }

      & > .a-button__icon--is-loading {
        position: absolute;
        height: auto;
        width: auto;
        opacity: 1;
      }
    }
  }

  &__icon {
    height: 100%;
    width: 40px;
  }

  &__text {
    flex: 1;
    display: flex;
    justify-content: center;
    align-items: center;
    &--icon {
      margin-right: 32px;
    }
  }
}
</style>
