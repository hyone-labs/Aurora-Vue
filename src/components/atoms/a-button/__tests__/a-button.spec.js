import { render, fireEvent } from '@testing-library/vue';
import AButton from '../a-button.vue';

describe('AButton', () => {
  it('render correctly', () => {
    const { getByText, debug } = render(AButton, {
      slots: {
        default: 'blah',
      },
    });

    debug();
  });
});
