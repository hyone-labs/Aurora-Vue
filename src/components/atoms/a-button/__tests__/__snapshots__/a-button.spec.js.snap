// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AButton render correctly with attrs 1`] = `
"<button class=\\"a-button a-button--behavior-is-rounded a-button--variant-primary a-button--size-small\\" id=\\"my-first-component\\" data-test-id=\\"secondary\\">
  <!--v-if-->
  <!--v-if--><span class=\\"a-button__text\\">With many Itens</span>
  <!--v-if-->
</button>"
`;

exports[`AButton render correctly with default props 1`] = `
"<button class=\\"a-button a-button--behavior-is-rounded a-button--variant-primary a-button--size-small\\">
  <!--v-if-->
  <!--v-if--><span class=\\"a-button__text\\">My first test</span>
  <!--v-if-->
</button>"
`;

exports[`AButton render correctly with different props 1`] = `
"<button class=\\"a-button a-button--behavior-block a-button--variant-secondary a-button--size-small\\">
  <div class=\\"a-icon a-icon--size-inherit a-icon--color-primary a-button__icon a-button__icon--position-left\\"><i class=\\"fas fa-circle\\"></i></div>
  <!--v-if--><span class=\\"a-button__text\\">With many Itens</span>
  <!--v-if-->
</button>"
`;
